### PR TITLE
referal code update, missing provider in user modal

### PR DESCRIPTION
**Pull request by ricardobrg**

### CODE REVIEW

**Comment by ricardobrg** [comment]
fix #215

File: lib/src/slices/data_screen/data_screen_service.dart [hunk 1]
```diff
@@ -19,8 +20,10 @@ class DataScreenService extends ChangeNotifier {
   late final DataScreenController controller;
   final DataBkgService _dataBkgService;
   final ApiGoogleService _googleService;
+  final ApiAppDataService appDataService;
```
Comment (mike-audi) [inline lib/src/slices/data_screen/data_screen_service.dart:23]: appDataService doesn't look like it's used in this class. revert this change

File: lib/src/slices/login_flow/login_flow_service.dart [hunk 2]
```diff
@@ -3,6 +3,7 @@
  * MIT license. See LICENSE file in root directory.
  */
 
+import 'package:app/src/slices/api_email_sender/api_email_sender_service.dart';
```
Comment (mike-audi) [inline lib/src/slices/login_flow/login_flow_service.dart:6]: use relative path

File: lib/src/slices/home_screen/ui/home_screen_layout.dart [hunk 3]
```diff
@@ -41,8 +41,9 @@ class HomeScreenLayout extends StatelessWidget {
               apiEmailSenderService: apiEmailSenderService,
               apiCompanyService: apiCompanyService,
               apiAppDataService: appDataService),
-          dataScreenService: DataScreenService(googleService, dataBkgService),
-          walletScreenService: WalletScreenService(),
+          dataScreenService:
+              DataScreenService(googleService, dataBkgService, appDataService),
```
Comment (mike-audi) [inline lib/src/slices/home_screen/ui/home_screen_layout.dart:45]: revert this line, since dataScreenService doesn't require appDataService

File: lib/src/slices/user_account_modal/user_account_modal_controller.dart [hunk 4]
```diff
@@ -6,6 +6,7 @@
 import 'package:app/src/slices/api_signup/api_signup_service.dart';
 import 'package:app/src/slices/login_flow/login_flow_service.dart';
 import 'package:app/src/slices/user_account_modal/user_account_modal_service.dart';
+import 'package:app/src/slices/user_referral/user_referral_service.dart';
```
Comment (mike-audi) [inline lib/src/slices/user_account_modal/user_account_modal_controller.dart:9]: use relative path

File: lib/src/slices/user_referral/model/user_referral_model.dart [hunk 5]
```diff
@@ -5,4 +5,6 @@
 
 class UserReferralModel {
   int referCount = 0;
+  String? code;
+  bool referCountUpdated = false;
```
Comment (mike-audi) [inline lib/src/slices/user_referral/model/user_referral_model.dart:9]: not needed

File: lib/src/slices/user_referral/ui/user_referral_view_code.dart [hunk 6]
```diff
@@ -36,8 +35,7 @@ class UserReferralViewCode extends StatelessWidget {
             Container(
                 margin: EdgeInsets.only(
                     left: 2.w, top: 1.5.h, right: 1.h, bottom: 1.5.h),
-                child: Text(
-                    service.getCode(Provider.of<LoginFlowService>(context)),
+                child: Text(service.model.code ?? service.getCode(),
```
Comment (mike-audi) [inline lib/src/slices/user_referral/ui/user_referral_view_code.dart:38]: shouldn't this just call service.getCode() which if it has the code it returns if it doesn't it shows "", otherwise this could get stuck in an infinite loop of

File: lib/src/slices/user_referral/user_referral_service.dart [hunk 7]
```diff
@@ -17,42 +19,56 @@ class UserReferralService extends ChangeNotifier {
...
   }
 
-  String getCode(LoginFlowService loginFlowService) {
-    String? code = loginFlowService.model.user!.user!.code;
-    if (code == null || code.isEmpty) _updateCode(loginFlowService);
+  String getCode() {
+    String? code = this.model.code;
+    if (code == null || code.isEmpty)
+      _updateCode(this.apiAppDataService, this.loginFlowService)
+          .then((_) => updateReferCount());
```
Comment (mike-audi) [inline lib/src/slices/user_referral/user_referral_service.dart:38]: this works, but if you wanted you could also just make it an async func with an await

File: lib/src/slices/user_referral/user_referral_service.dart [hunk 8]
```diff
@@ -17,42 +19,56 @@ class UserReferralService extends ChangeNotifier {
...
-    if (HelperApiUtils.isOk(rsp.code)) {
-      ApiBlockchainModelAddressRspCode data = rsp.data;
-      loginFlowService.model.user!.user!.code = data.code;
-      await loginFlowService.apiUserService
-          .setUser(loginFlowService.model.user!.user!);
-      await loginFlowService.loadUser();
-      notifyListeners();
+  Future<void> _updateCode(ApiAppDataService apiAppDataService,
+      LoginFlowService loginFlowService) async {
+    String code =
```
Comment (mike-audi) [inline lib/src/slices/user_referral/user_referral_service.dart:57]: shouldn't this just be a nullable?

**Review by mike-audi — CHANGES_REQUESTED** [review]
I've added several changes to files. In general the code looks great, just some very minor clean up and then we can merge

**Review by mike-audi — CHANGES_REQUESTED** [review]
Pass ApiBlockchainService directly to ReferralService methods instead of using a getter in the LoginFlowService.

**Review by mike-audi — APPROVED** [review]
